Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -1097,6 +1097,8 @@ public boolean playLand(Card card, Game game, boolean ignoreTiming) {
         if (!ignoreTiming && !playLandAbility.canActivate(this.playerId, game)) {
             return false;
         }
+        
+        game.fireEvent(GameEvent.getEvent(GameEvent.EventType.PLAY_LAND, card.getId(), card.getId(), playerId));
```

Comment:
Why it's here? 
1. It's have another PLAY_LAND event below too (e.g. it's can raise twice).
2. It's have another code to ignore land cast (e.g. it's can raise event, but real land will not casted).

Reply:
To fire an event and to check for replacement of an event is not the same.

Often we do such three steps:
1) Fire Event (PLAY_LAND)
2) Check if event is replaced (PLAY_LAND)
3) Fire that the event really happened (LAND_PLAYED)

Without checing the code further, it seems like that for PLAY_LAND step one was not implemented yet, because there was no need.